discretisedfield/field.py

```diff
@@ -3475,3 +3476,117 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+
+        units : str, optional
+
+            String to set units of the field ``DataArray``
+
+        Returns
+        -------
+        xarray.DataArray
+
+            Field values DataArray.
+
+        Raises
+        ------
+        TypeError
+
+            If either ``name`` or ``units`` argument is not a string.
+
+        Examples
+        --------
+        >>> import discretisedfield as df
+        ...
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (10, 10, 10)
+        >>> cell = (1, 1, 1)
+        >>> mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+        >>> field = df.Field(mesh=mesh, dim=3, value=(1, 0, 0), norm=1.)
+        ...
+        >>> field
+        Field(...)
+        ...
+        >>> field.to_xarray()
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10, comp: 3)>
+        array[...]
+        Coordinates:
+        * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * comp     (comp) <U1 'x' 'y' 'z'
+        ...
+        >>> field.to_xarray().sel(comp='x')
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10)>
+        Coordinates:
+        * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        comp     <U1 'x'
+        ...
+        >>> field.to_xarray(name=21)
+        Traceback (most recent call last):
+        ...
+        TypeError: name argument must be a string.
+        >>> field.to_xarray(units={'field': 'A/m'})
+        Traceback (most recent call last):
+        ...
+        TypeError: units argument must be a string.
+
+        """
+
+        if type(name) != str:
+            msg = "name argument must be a string."
+            raise TypeError(msg)
+        elif units is not None and type(units) != str:
+            msg = "units argument must be a string."
+            raise TypeError(msg)
+
+        geo_dim = ['x', 'y', 'z']
+
+        data_array_coords = {dim: np.fromiter(self.mesh.axis_points(dim),
+                                              dtype=float)
+                             for dim in geo_dim
+                             }
+
+        geo_units_dict = {dim: 'm' for dim in geo_dim}  # fix units for geo
+
+        if self.dim != 1:
+            data_array_dims = geo_dim + ['comp']
+            data_array_coords['comp'] = self.components
+            field_array = self.array
+        else:
+            data_array_dims = geo_dim
+            field_array = np.squeeze(self.array, self.array.ndim - 1)
+
+        data_array = xr.DataArray(field_array,
+                                  dims=data_array_dims,
+                                  coords=data_array_coords)
+
+        for dim in geo_units_dict:
+            data_array[dim].attrs['units'] = geo_units_dict[dim]
```

Comment:
`unit` without `s`?

Reply:
The `units` attribute comes from netCDF convention that xarray follows as mentioned [here](https://xarray.pydata.org/en/stable/getting-started-guide/quick-overview.html#attributes). xarray uses this attribute mostly as plot labels.

Reply:
I wasn't aware of this standard. We should keep `units` everywhere.